### PR TITLE
refactor(skipLast): fix names, update docs and comments

### PR DESCRIPTION
**Pull request by benlesh**
- Fixes names to be a little more relevant.
- Adds additional comments to the code about why it is doing what it is doing
- Updates documentation
- Remove incorrect documentation about thrown values

Inspired by @jakovljevic-mladen's PR: #5784

### CODE REVIEW

**Comment by jakovljevic-mladen** [comment]
Hi @benlesh, thank you a lot for this PR and for taking care of it. I will gladly take a look at this PR.

File: src/internal/operators/skipLast.ts [hunk 1]
```diff
@@ -5,76 +5,92 @@ import { operate } from '../util/lift';
...
+ * Skip the last 2 values of an observable with many values
+ *
  * ```ts
- * import { range } from 'rxjs';
+ * import { of } from 'rxjs';
  * import { skipLast } from 'rxjs/operators';
  *
- * const many = range(1, 5);
+ * const numbers = of(1, 2, 3, 4, 5);
  * const skipLastTwo = many.pipe(skipLast(2));
```
Comment (jakovljevic-mladen) [inline src/internal/operators/skipLast.ts:33]: `many` should be renamed as well to `numbers`.

File: src/internal/operators/skipLast.ts [hunk 2]
```diff
@@ -5,76 +5,92 @@ import { operate } from '../util/lift';
...
+ * Skip a specified number of values before the completion of an observable.
  *
  * ![](skipLast.png)
  *
- * `skipLast` returns an Observable that accumulates a queue with a length
- * enough to store the first `count` values. As more values are received,
- * values are taken from the front of the queue and produced on the result
- * sequence. This causes values to be delayed.
+ * Returns an observable that will emit values as soon as it can, given a number of
+ * skipped values. For example, If you `skipLast(3)` on a source, when the source
```
Comment (jakovljevic-mladen) [inline src/internal/operators/skipLast.ts:13]: `If` should be `if`.

File: src/internal/operators/skipLast.ts [hunk 3]
```diff
@@ -5,76 +5,92 @@ import { operate } from '../util/lift';
...
+ * as each value is held in a buffer until enough values have been emitted that that
+ * the buffered value may finally be sent to the consumer.
+ *
+ * After subscribing, unsubscribing will not result in the emission of the buffered
+ * skipped values.
  *
  * ## Example
- * Skip the last 2 values of an Observable with many values
+ *
+ * Skip the last 2 values of an observable with many values
```
Comment (jakovljevic-mladen) [inline src/internal/operators/skipLast.ts:26]: Maybe add `:` or `.` at the end of sentence.

**Review by jakovljevic-mladen — COMMENTED** [review]
LGMT. 2 small nitpicks and an error in an example. I think that you solved the issue that I tried much better than I have. And updating docs and adding comments is also 👍

File: src/internal/operators/skipLast.ts (hunk 3 above)
Comment (jakovljevic-mladen) [reply src/internal/operators/skipLast.ts:26]: Now I see that many examples in many other files don't have none of what I proposed here.

**Comment by cartant** [comment]
@benlesh Is this ready to be reviewed?

**Review by cartant — APPROVED** [review]
Rebased it. LGTM.

**Pull request merged**